clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -185,8 +186,11 @@ func getIsPodUpdatedPodPredicateFromUnstructured(obj *unstructured.Unstructured)
 		return nil, goerrors.Errorf("failed to parse spec.teemplate as v1.PodTemplateSpec")
 	}
 
-	return func(pod *corev1.Pod) bool {
-		return equality.Semantic.DeepDerivative(template.Spec, pod.Spec)
+	return func(pod *corev1.Pod) (bool, error) {
```

Comment:
Can we return only error where nil means that are equal?

Otherwise, how caller should handle cases when e.g. (true, non-nil) or (false, nil)?

Reply:
That's a good point. And this will simplify a code in few places. Let me implement this change.

Reply:
Done.